Framework/Algorithms/test/ConvertSpectrumAxis2Test.h

```diff
@@ -391,4 +391,34 @@ class ConvertSpectrumAxis2Test : public CxxTest::TestSuite {
   }
 };
 
+class ConvertSpectrumAxis2TestPerformance : public CxxTest::TestSuite {
+public:
+  static ConvertSpectrumAxis2TestPerformance *createSuite() {
+    return new ConvertSpectrumAxis2TestPerformance();
+  }
+  static void destroySuite(ConvertSpectrumAxis2TestPerformance *suite) {
+    delete suite;
+  }
+
+  ConvertSpectrumAxis2TestPerformance() {}
+
+  void test_conversion_to_signed_theta_with_many_entries() {
+    std::string inputWSName("inputWS");
+    std::string outputWSName("outputWS");
+
+    auto testWS = WorkspaceCreationHelper::create2DWorkspaceWithFullInstrument(
+        20000, 20000);
+    AnalysisDataService::Instance().addOrReplace(inputWSName, testWS);
+
+    Mantid::Algorithms::ConvertSpectrumAxis2 conv;
+
+    conv.initialize();
+    conv.setPropertyValue("InputWorkspace", inputWSName);
+    conv.setPropertyValue("OutputWorkspace", outputWSName);
+    conv.setPropertyValue("Target", "SignedTheta");
+    conv.setPropertyValue("EFixed", "10.0");
+    conv.execute();
+  }
```

Comment:
Am i missing something or is this failing to test anything? No expectations/asserts.